sirepo/job_supervisor.py

```diff
@@ -809,3 +854,75 @@ def send(self):
 
     def __hash__(self):
         return hash((self.opId,))
+
+
+def _purge_free_simulations_set():
+    def tomorrow_2am_mst():
+        if pkconfig.channel_in_internal_test():
+            return sirepo.srtime.utc_now_as_float() \
+                + cfg.test_purge_frequency_secs
+        # 9am UTC == 2am MST: Does not adjust for daylight savings
+        t = sirepo.srtime.utc_now() + datetime.timedelta(days=1)
+        return sirepo.srtime.to_timestamp(
+          datetime.datetime(t.year, t.month, t.day, 9, 0, 0, 0)
+        )
+
+    tornado.ioloop.IOLoop.current().call_at(
+        tomorrow_2am_mst(),
+        _purge_free_simulations,
+    )
+
+
+async def _purge_free_simulations():
+    async def _get_uids_and_files():
+        r = PKDict()
+        p = sirepo.auth_db.UserRole.uids_of_paid_users()
+        for f in pkio.sorted_glob(_DB_DIR.join('*')):
+            u = str(f.basename).split('-')[0]
+            if u in p or not _is_older_than_days(f):
+                continue
+            if u not in r:
+                r[u] = []
+                await tornado.gen.sleep(0) # TODO(e-carlin): necessary?
+            r[u].append(f)
+        return r
+
+    def _get_run_dir_path(data):
+        data.report = data.computeModel
+        return sirepo.simulation_db.simulation_run_dir(data)
+
+    def _is_older_than_days(db_file):
+        return db_file.mtime() < (
+            sirepo.srtime.utc_now_as_float() -
+            (cfg.purge_free_after_days * 24 * 60 * 60)
+        )
+
+    def _purge_sim(db_file):
+        # OPTIMIZATION: We assume the uids_of_paid_users doesn't change very
+        # frequently so we don't need to check again. A user could run a sim
+        # at anytime so we need to check that they haven't
+        if not _is_older_than_days(db_file):
```

Comment:
At this point, we want to check the value inside the db_file, because the db file might be updated for any reason, e.g. db upgrade, but the true test is to validate the computeJobStart. The use of mtime is an efficiency hack to trim a large list quickly to likely candidates.

Reply:
What about using lastUpdatetime? It would be closer to mtime and I think closer to "last time of interaction."

Reply:
Is lastUpdateTime always set? If so, yes.